### ISSUE TITLE
Zotino v1.1

### ISSUE THREAD

**Issue by hartytp** [m-labs/sinara#446]
- [x] Remove simulations page from schematic (let's just leave the Spice files in the repository)
- [x] Update power budget
- [x] Use two output resistors in series/parallel to improve power handling (200mW short-circuit)
- [x] reverse C143, C145, C147, C84
- [x] mirrored SOT23 diode footprint due to Altium bug - part of component is on top layer, pads are on bottom layer.
- [x] add Shottky diodes at the LDO outputs
- [x] 3mm higher EMC shield on PSU
- [ ] hole for heatsink in EMC shield
- [x] reverse IDC connector on BNC to IDC board
- [ ] Zero-index all channels
- [x] swap BNC connector GND terminals
- [x] either add ~2mm plastic washers on BNC barrel **behind** front panel or move BNC connectors 2mm towards FP. This is to enable fixing the BNC connectors to the panel using the screws.
- [x]  Please remove the annotation "This module connects to Kasli ..." and replace it with something like "EEM connector: IO are LVDS, I2C is 3V3 LVCMOS, P3V3_MP up to 20mA, P12V up to 1A." (use same annotation as other EEMs)
- [x] run PI and thermal simulations. Please can you also double check that all regulators can handle the required current etc. Please also double check the layout for any places that could cause issues (bad cross-talk or patristics that could lead to oscillation, etc)
- ~Can we run the digital interface from 2V5?~ No -- see sampler discussions about this!
- [x] Fix temperature controller issues:
- [x] Silk + schematic + panel should be "v1.x" not "rev1.x" Silk should just say "Zotino v1.1" (copy from Sampler)  *still needs doing: Add Zotino name to schematic and pcb, add version number to schematic and front panel*
- [x] Only have one set of +-12V regulators, not two!
- [x] Reduce SMPs outputs slightly, since we're only using +-12 regulators
- [x] Remove BNC to IDC connectors from this; they should be a separate project (they are a useful adapter for other boards like Sampler, and not part of the DAC design).
- [x] Upload BNC to IDC connector boards as a separate project.
- [x] Update front panel (see discussion in other issues)
- [x] Let's remove the simulations page from the schematic, better to just include the simulation itself in the project directory
- [x] Tidy up adapter boards cf comments [here](https://github.com/m-labs/sinara/commit/42682834eb78feccc2dbdd3f7108e2088076cef6#commitcomment-26854301) (two posts). Add annotations on silk and schematic for the adapter boards indicating how the connectors map to the channel (e.g. top=0, bottom=1 for the BNCs etc).
- [x] Remaining temperature controller issues [here](https://github.com/m-labs/sinara/issues/446#issuecomment-356663333)
- [x] Add more test points for power supply rails (both at the SMPS and for the LDOs), SPI, etc
- [x] For the TEC IDC, let's add a ground wire between the TEC and the thermistor to reduce EMI from the noisy TEC current (which is rectified by the temperature sensor, leading to errors).

**Comment by dhslichter** [comment]
I would vote strongly for removal of the thermal island and instead using vias/ground planes to spread heat from the DAC and make it equilibrate better to the board.  This ought to keep the DAC from running itself to death temp-wise, at least.

**Comment by hartytp** [comment]
@gkasprow Happy New Year!

Sorry to add to the huge to do list, but is there any chance we can put this in the pipeline to do asap?

This is very important to us because we don't currently have a version of this board that we can use for testing (the current version kills the DACs too quickly to be testable). So, I'd like to get a version with bug fixes sent to manufacture as soon as possible.

Hopefully these changes are small and will be quite quick to implement.

Obviously, crucial Sayma/RF PA issues (like the 1V8 diagnostics) should take priority.

**Comment by hartytp** [comment]
Greg, is there any chance we can get this sent off to manufacture this week? I'd really like to get this done so we can start testing properly.

For the temperature control stuff, here are my thoughts:
- Let's scrap the thermal island
- Add some way for users to mount a TEC. Do this however you think is best, but here is my initial thought (feel free to overrule)
  - Leave some exposed copper on the board's underside along with some mounting holes that can be used to attach a peltier and heat sink (I'm assuming that it's better to temperature control the ground that the DAC and reference connect to than to try to mount the TEC directly on the DAC, but feel free to do what you think is best here)
  - Run a thermal simulation to verify that a TEC which can drive +-3W into the thermal island is able to change the DAC's temperature by +-2C or so (that should be enough to ensure regulation in most lab conditions once the board has warmed up)
  - Users who want the best stability can attach a self-adhesive insulating strip over the DAC + reference on the top side of the board to scree out air currents
  - Add a 10k NTC SMT thermistor on the board's underside as close the the TEC as possible
  - Bring the 10k thermistor to a header on the top of the board. This header should ideally also connect to the TEC (are there any TECs that have SMT connections?). The header can then be used to connect this to an external temperature controller EEM via ribbon cable/IDC (see the Thermostat EEM page on the wiki for an idea of what this could look like)
  - Scrap all temperature controller related components on the PCB (including the LEDs)

How does that sound?

Thanks!

**Comment by hartytp** [comment]
Greg, thanks for posting the initial v1.1 schematics! Can you do me a favour, please and tick off the items you've addressed on the list at the top of this post (that will make my design review easier).

**Comment by hartytp** [comment]
Greg, I've checked over the schematic you uploaded and it mainly looks good. Apart from the outstanding issues above (I've ticked off the ones that looked fixed to me):

- Did you run thermal + PI simulations yet? In particular, I'd be interested to see how many C/W a peltier would produce here so that we know we can temperature stabilise this with a sensible TEC power.
- To match the peltier to the I/V rating of the proposed Thermistor EEM, I'm assuming the TEC we'll use will be something like [this](http://uk.farnell.com/multicomp/mcpf-031-10-25/peltier-cooler-4-5w/dp/1639756)
  - What is the diameter of the holes you used for the TEC wires? Will they be wide enough for this kind of TEC? (I didn't see that specified on the data sheet)
  - [ ] Please can you add an annotation on the schematic and silk screen for the polarity of the TEC wires ("TEC+" and "TEC-"). Please name the nets TEC+ and TEC- instead of B1 and B2
  - [ ] The TEC wiring should be able to handle at least 2A with no problems (and without too much voltage drop in the wiring harness). Should we use 1 or 2 more pins for the TEC to ensure that's not a problem? Would it be worth alternating TEC+ and TEC- in the ribbon cable to reduce emission from the TEC current? (Or is that overkill?)
  - [ ] Is the exposed copper area large enough to mount a 15mm x 15mm TEC?
- [ ] After thinking about this more, let's remove the P12V0, P3V3_MP and I2C lines from the temperature controller header. They're not really necessary with the proposed temperature controller EEM design. (This presumably means we can remove the I2C extender from the design, which is nice!)
- [ ] Do the cutouts you've left in around the DAC make any difference to the thermal management? Or, is it better to remove them completely?
- [ ] Can you add a hole in the screening can for the TEC + heat sink?
- [ ] would it make sense to design the heat sink mounting holes around some standard commercially available heat sink hole pattern. I'd like it if we don't have to machine the heat sink to make it work. We should also include a suggested heat sink PN in the schematic
- Are there any disadvantages of having the DAC on the underside of the PCB? I guess we already have quite a few components there, so this isn't a problem (won't make the board more expensive or more fragile than it already is).
- Since the DAC is on the underside of the PCB, there isn't room to add insulating foam on top of the DAC (unless one also adds a filler panel). That should be fine since the DAC's thermal paddle means it is very well heat sunk to the PCB, so small air fluctuations won't be too much of an issue (and we're only aiming for +-200mK or so stability, which isn't too tough to meet).

Other than that, looks good to me!

**Comment by hartytp** [comment]
Oh, one other thing: let's wait and see what @cjbe's measurements tell us about the noise on Zotino v1.0 before doing anything about this, but...

The SMPS we're currently using has over current protection, but not over voltage or over temperature protection AFAICT. I wonder if it's a little fragile, and if some of the abuse we've given it could have lead to the SMPS failing on some of our boards?

**Comment by hartytp** [comment]
@gkasprow IIRC, for cross-talk testing, we were planning to build a trivial board which breaks out a HD68 to 4 x IDCs. Did we ever do that? I can't find it in the repository. If not, is it possible to do that for Zotino v1.1? Thanks!

**Comment by gkasprow** [comment]
@hartytp The HD68 to IDC  and IDC to BNC adapters were added to repo

**Comment by hartytp** [comment]
Thanks!

**Comment by a-shafir** [comment]
How about separate with beads the P/N 12V for each load (DAC, and each opamp)? There will be some AC leakage via D5A/B capacitance etc.  Also R3 can be at least <100R safely.

I think additional LC filter makes sense to add for the DAC DVCC line at least to reduce the return AC current via C83, C13, C14 to the power supply that can leak to the analog side.

I think it makes sense add additional inductor similar to L8 and power IC1 and IC35 separately so it will reduce cross-interference. 

It makes sense to route the IC3 and related parts ground directly to IC18 AGND pins (star routing) but not to the gnd plane. Also it makes sense to have the ground plane between the front panel side cut under most of the boards and have connected only under the DAC so it will prevent parasitic digital ground noise.

I can check the layout as well. According the DAC d/s it is quite sensitive for the layout and grounding.

**Comment by hartytp** [comment]
@a-shafir 

> How about separate with beads the P/N 12V for each load (DAC, and each opamp)?

I don't believe that a ferrite bead on each OpAmp would be helpful; SMT ferrites typically only work for 10s of MHz and above, but this circuit only works at audio frequencies so their affect will be minimal. I also do not believe that a ferrite between the DAC's analog supply rails and the OpAmps is helpful for the same reason.

Note that there is already heavy LC filtering between the SMPS and the LDOs.

All digital circuitry (which has frequencies that could benefit from a ferrite bead) is already run from a different regulator to the analog circuitry (it's 3V3).

> There will be some AC leakage via D5A/B capacitance etc.

Really? The capacitance for those diodes is of order 1pF. That forms a pole at 300MHz with the 500Ohm output impedance. Again, given that this circuit is only designed to operate at audio frequencies, and that the supply rails are very low noise, that will not be a problem. Moreover, the  DAC output capacitor is 2.2nF, a thousand times larger and so will filter off any small noise coupled from the supply rails.

> Also R3 can be at least <100R safely.

It potentially could be, but there should be no need for that; those regulators should be very low noise already. Let's wait until we start making noise measurements and only change this if we find a problem.

> I think additional LC filter makes sense to add for the DAC DVCC line at least to reduce the return AC current via C83, C13, C14 to the power supply that can leak to the analog side.

I don't understand what you mean here. You want to put an inductor between the DAC and the decoupling capacitors to prevent digital currents returning to ground through those caps? That's not a good idea. This supply is absolutely fine IMHO and not going to cause any problems.

Let's leave this as it is for now. We will test the prototypes carefully for noise issues and make changes are required.

> I think it makes sense add additional inductor similar to L8 and power IC1 and IC35 separately so it will reduce cross-interference.

Really not necessary. If a small amount of noise couples from the input of one switch mode supply to the input of another that won't be a problem -- a tiny amount of cross-talk between the supplies won't degrade their overall performance and they are already heavily filtered on their outputs.

> It makes sense to route the IC3 and related parts ground directly to IC18 AGND pins (star routing) but not to the gnd plane. 

No, I'd prefer not to do this. These kinds of ground plane cuts are rarely a good idea. Better to do this with a good layout that keeps the reference near the DAC and ensures that both share a low impedance ground-plane.

> Also it makes sense to have the ground plane between the front panel side cut under most of the boards and have connected only under the DAC so it will prevent parasitic digital ground noise.

No! Again, these kinds of ground cuts are generally not a good idea and usually cause more problems than they solve.

**Comment by hartytp** [comment]
@a-shafir I appreciate the enthusiasm for improving these boards. But, for now I don't want to make those kinds of changes. Your approach would overcomplicate these designs by adding components to try to solve things that are unlikely to be issues.

Let's just wait until we have the next prototype. Once we've characterised that thoroughly we can see what the actual performance is and evaluate changes based on that. If the performance is bad then we can consider making a new version with some of these modifications.

**Comment by a-shafir** [comment]
Ok, about the ground plane this is exactly type of the circuit where it works perfectly. The digital part of the board is quite noisy and imho there is no reasons to spread the return currents via whole board ground/power planes.
Even this looks like over-caution it certainly will not harm in this board where there is only one A-D junction but might help in some case. It cost nothing including the design. There is no reason to afraid the "analog ground" - it only helps in most of the cases. I am not suggesting a full ground separation with beads etc - this indeed can be dangerous in many cases.

Basically i see only not well under control noise source - the ground. Including the VREF signal path.

**Comment by hartytp** [comment]
> Some simple numbers: reactive impedance of a trace 100mm wide 0.5oz copper is ~0.000247Ohm so 1A DC current will be 0.24mV drop. The DAC resolution on the output roughly 10V/2^16 ~= 0.15mV. So it is in the same scale. With AC it worst, roughly 10 times at 77KHz.

Firstly, this statement makes no sense. Reactive impedance of a trace at DC? You mean resistive? Where do you get 1A from? What kind of digital logic has that kind of current? I appreciate the effort to quantify your statements, but this calculation is irrelevant.

> Even this looks like over-caution it certainly will not harm in this board where there is only one A-D junction but might help. 

I disagree with this statement. It's really easy to take a perfectly good board and ruin it by splitting ground planes.

Also, note that there is a pretty aggressive output filter on these boards that will make a large difference to any high-frequency digital noise.

---

As the one funding this project, here is an executive decision on this matter to avoid wasting time on it:
unless @gkasprow says it's necessary, we will not do this for the prototypes. If there is a noise issue (beyond the expected clock feedthrough, LDAC glitches etc) with them then we will reconsider in a future revision

---

> BTW: in case of using peltier cooler it makes sense to have proper cleaning and possible conformal coating in the affected area because the it leads to condensation etc.

If you're getting condensation on the peltier here then you're doing something wrong.

**Comment by hartytp** [comment]
@a-shafir re diode capacitance. Note that the regulators we're using on this board [negative LOD data sheet](http://cds.linear.com/docs/en/datasheet/1964fb.pdf) [positive LDO datasheet]() have noise floors below 100nV/rtHz. So, for higher frequencies, that's divided by 1000:1 by the output capacitor, giving a noise floor for the board below 100pV/rtHz.

At lower frequencies the supply noise is larger, but it's shunted more effectively by the 500R output resistor.

So, unless there are some large spurs on the regulator (e.g. leakage due to the SMPS) I do not expect an issue. If we see those then we will definitely add an LC filter/ferrite bead/etc.

**Comment by hartytp** [comment]
@gkasprow On this subject: @a-shafir's posts did make me look again at the supply network (so, thanks for that) so I noticed that the negative LDO we're using is the version without the bypass pin. As this is quite a noise sensitive device, we should probably use the bypassable version unless there is a good reason no to (e.g. big cost difference). Why didn't we use it?

NB If we change this on Zotino, we should also change it on all other boards using this regulator (is that just sampler?).

**Comment by hartytp** [comment]
@gkasprow I had more of a look at the measurement setup we were using here to look at the noise. Based on my new measurements, I currently believe that the noise @cjbe saw is related to the measurement setup and not to the DAC itself.

So, I don't think there is any need to do further testing on the v1.1 boards.

**Comment by jordens** [comment]
@Hartytp Care to elaborate?

**Comment by hartytp** [comment]
I think the bulk of what Chris was seeing was pickup/grounding/wiring issues.

I'm now using a small coax pigtail soldered directly to the output of the CMC on Zotino to do my probing (DAC removed, OpAmp input grounded). I'm attaching that to a low noise FFT analyzer. I measure a noise floor of something like 17nV/rtHz with lots of junk on it at higher frequencies (instrument noise floor is around 5nV/rtHz). However, AFAICT, all that high frequency junk is just pickup (I see a similar spectrum, but with at a much larger level with the board unpowered). e.g. the noise floor depends more on whether the PSU is plugged in (output off) than whether the board is actually powered.

But, these are the standard low-noise measurement issues one always sees. Doing this measurement properly isn't that hard, just a bit time consuming (battery powered pre-amps, careful wiring and grounding etc) and I don't have time to do it right now.

My feeling is that we should do this all properly once we have the v1.2 design and not spend any more time on the current boards.

**Comment by hartytp** [comment]
@jordens Maybe I'll regret saying this, but I'm not too worried about the filter design, which is the only thing I can test at the moment (e.g. can't test the DACs because they die too fast) so I think it makes sense to move onto v1.2 without further delay.

The filter itself is just an OpAmp circuit. We've simulated it in SPICE and it looks good. But even if there is a problem there, I'm confident that we can make it work by changing component values/OpAmps. So, the only real danger is noise sources external to the filter, such as:
- The DACs/digital stuff: this could be a serious issue, but I can't test it at the moment.
- The SMPSs: from Sampler, we know that these SMPSs can be used in very low noise designs without issue; that design achieved 8nV/rtHz (consistent with the InAmp noise floor) and if we had to, we could live with that for this design since we will generally have a filter by the trap which can provide the final 20dB.
- External pickup: not much we can do about that apart from add external screening (the layout is already pretty good and we do already have CMCs etc).

**Comment by gkasprow** [comment]
@hartytp the TEC wires holes are 1.1mm. Should be fine.
IDC contact current is usually rated 1A. I added 2 more pins and bigger IDC connector.
Do we really need 15x15mm TEC? I was thinking about 9x9 one...
The I2C extender is also routed to the DAC reset pin. I'm not sure if anyone is going to use that.
OK, let's remove cutouts and make more space for TEC and mounting
I changed the LDO to LT1964ES5-BYP#TRMPBF
There are already cut-outs around theSMPS to limit the return current

**Comment by gkasprow** [comment]
It's not easy to find small heatsink with mounting holes...

**Comment by hartytp** [comment]
> It's not easy to find small heatsink with mounting holes...

What do you want to do then? Glue it? Or, are there PCB mount clips we can use to hold it in place? I don't mind what you do here so long as there is some easy way of mounting it.

**Comment by gkasprow** [comment]
TEC modules need quite a lot of force to work properly.
Let's use [this](https://pl.mouser.com/ProductDetail/Advanced-Thermal-Solutions/ATS-53170D-C1-R0?qs=sGAEpiMZZMttgyDkZ5WiulqfIpz7MfclvtavMunAIpw%3d) type of thing. I use bigger version of them. The yellow frame will hold Peltier module, I will add PCB-mounted hooks for the springs

**Comment by hartytp** [comment]
@gkasprow That looks good. I'd be happy to go for something like that.

Please can you add a suitable TEC and heatsink to the BOM? It would be good to populate this for the test boards so that we can check it all works during prototyping. (For that I would probably use the evaulation board that Thorlabs sell, which hooks up to USB and does PID + peltier driver).

**Comment by gkasprow** [comment]
I mean such jumpers that are sold as components and will manage to keep such assembly
![obraz](https://user-images.githubusercontent.com/4325054/35363108-b4711bb6-0168-11e8-9069-5939294684c3.png)

**Comment by hartytp** [comment]
@gkasprow That sounds good. I just mean that it would be good to also choose a TEC + heatsink + clip and attach it to the PCB. That checks it all fits together and also allows us to make measurements like temperature stability of the DAC more easily.

**Comment by gkasprow** [comment]
sure, let's add it to the BOM. I will also make mechanical models to make sure it fits.

**Comment by hartytp** [comment]
great!

**Comment by gkasprow** [comment]
@jordens @hartytp we can leave I2C, then add I2C <-> UART [bridge chip ](https://www.nxp.com/docs/en/brochure/75015676.pdf)and control TEC controller from Zotino board directly

**Comment by gkasprow** [comment]
@hartytp Such [TEC](https://pl.mouser.com/ProductDetail/CUI/CP60139H?qs=sGAEpiMZZMtAhbGwPnfBjWfeDQfrfxJO0OeY1drMFSAjpdSSNUqxAA%3d%3d) should do the job..

**Comment by gkasprow** [comment]
Edit - I mean 10x10 modules, 15x15 are slightly too big.

**Comment by gkasprow** [comment]
but 15x15 are much cheaper...

**Comment by gkasprow** [comment]
OK, I will fit 15x15 as well. What about 2.6V module?

**Comment by gkasprow** [comment]
This [one](https://pl.mouser.com/ProductDetail/Laird/430511-504?qs=sRTx%2ftA2SD9NM5cdxVgMYw%3d%3d) should be fine

**Comment by klickverbot** [comment]
(@gkasprow: "This one" – did you mean to add a link?)

**Comment by hartytp** [comment]
>but 15x15 are much cheaper...

Yes, AFIACT, 15x15 is standard so it's best to go for that if we can.

> What about 2.6V module?

Let's go for one with closer to 2.6Ohm so it works with the nice [Maxim TEC driver](https://datasheets.maximintegrated.com/en/ds/MAX8520-MAX8521.pdf) (1.5A, 4V)

> @jordens @hartytp we can leave I2C, then add I2C <-> UART bridge chip and control TEC controller from Zotino board directly

Okay, I'm fine to leave the I2C there if it doesn't make much difference to the size of the connector. Even if Thermostat supports ethernet as its main interface, having an I2C interface as well would be nice. That way, it becomes just an extension of Zotino.

I guess we can mount the TEC driver board on a separate front panel adjacent to Zotino.

**Comment by jordens** [comment]
I think we can live without reset. If the power on sequence leads to a well defined state. The datasheet does not say much about the sequencing of the reset.
Since we are now looking at a TEC with heating and cooling, we can lower the thermal resistance between the DAC and the board as much as possible. The reason is to guarantee that this board will work well even without the TEC and associated logic. The only limit there is that the TEC needs to be able to heat away the all of the DAC power fluctuations times the ratio of the thermal resistances (ambient-TEC-DAC/DAC-board-ambient). That ratio should probably be above 0.5 given that the TEC can do +-2W and a cautious estimate for DAC power fluctuation of +-1W.

**Comment by hartytp** [comment]
> I think we can live without reset. If the power on sequence leads to a well defined state. The datasheet does not say much about the sequencing of the reset.

FWIW, I don't recall ever using the reset pin for these DACs. IIRC, I've only ever used the SPI interface to do everything and that's always been fine.

@jordens Do you need the BUSY pin (or CLR for that matter) on the EEM connector? One other option would be to take BUSY/CLR of the EEM connector and to put RESET directly on the EEM connector (more convenient than having it on I2C anyway).

----

In any case, as @gkasprow points out, it's probably worth keeping the I2C interface there for the temperature controller. If something like Thermostat is used to control Zotino, it's nice if it can be wired as basically being an I2C extension of Zotino (gains and setpoint programmed automatically on startup by Kasli) rather than needing an extra ethernet cable.

**Comment by hartytp** [comment]
> Since we are now looking at a TEC with heating and cooling, we can lower the thermal resistance between the DAC and the board as much as possible. The reason is to guarantee that this board will work well even without the TEC and associated logic. The only limit there is that the TEC needs to be able to heat away the all of the DAC power fluctuations times the ratio of the thermal resistances (ambient-TEC-DAC/DAC-board-ambient). That ratio should probably be above 0.5 given that the TEC can do +-2W and a cautious estimate for DAC power fluctuation of +-1W.

Agreed, that's the current plan. As @gkasprow said in a post above, he's removed the thermal island completely.

He's also planning to do a thermal simulation to check the C/W of the peltier and check that's it's something sensible (aiming for something like 1C/W).

**Comment by hartytp** [comment]
@gkasprow To confirm: in the initial prototypes we're planning to get Technosystems to populate the peltier stack (including heat sinks etc), right?

**Comment by gkasprow** [comment]
Yes.
I'm not sure if I manage to do the thermal simulation with TEC. I never did it before :)
I used simplified Hyperlynx tool for PCB level simulations.
For TEC simulation, I have to dig into the FloTherm package. I'll do it once we close issues that cannot wait.

**Comment by hartytp** [comment]
> I'm not sure if I manage to do the thermal simulation with TEC. I never did it before :)

Can you do this by adding a 1W resistor connected to ground where the TEC should go? See how much the DAC temperature changes with and without the resistor and that tells you everything you need to know.

**Comment by gkasprow** [comment]
This I can do. Maybe it's possible to specify negative power :) I will check

**Comment by hartytp** [comment]
You don't need to simulate negative power; the number of C/W will be about the same for positive and negative power. So, just simulate with a resistor delivering positive power and it will be fine! We just need a rough estimate of this to check that we're in the right ball park, we're not looking for something ultra-precise.

**Comment by jordens** [comment]
If your tools can't simulate that directly, we need to know two new thermal resistances. "junction-package-TEC" (from the DAC datasheet and a good number for the thermal contact of the TEC) and "TEC-ambient" (including the TEC heat sink). The third resistance "junction-board-ambient" is known and is what Greg has been simulating. Then with the efficiency P_thermal/P_electrical(delta T) of the TEC, whever wants to have a go at this can set up the thermal circuit.

**Comment by gkasprow** [comment]
I can add 1W of heat source, no problem.

**Comment by hartytp** [comment]
> I can add 1W of heat source, no problem.

Great! That gives enough information to check that this will work.

Remember that we only need to take our DAC's temperature fluctuations from a few degress to a few hundred mK, so this isn't a very demanding application. As a result, I'm happy ignoring things like the DAC-ambient thermal impedance (if this is an issue then we can glue insulating foam over the DAC).

**Comment by jordens** [comment]
@gkasprow Just out of curiosity: The four mounting holes in each EEM PCB that have so far been unused, do they by chance fit the slots for nuts and screws in [these heat sinks](http://www.fischerelektronik.de/fileadmin/fischertemplates/download/Katalog/gehaeuse.pdf#page=120)?
![image](https://user-images.githubusercontent.com/1338946/35399899-b3afd27a-01f5-11e8-9ce9-3f0a8a91bbd4.png) 

With a spacer between PCB underside and heat sink to make some clearance for SMD parts that could be a simple solution to cooling certain parts of these EEMs, like Urukul or Zotino.

**Comment by gkasprow** [comment]
@jordens these heat sinks have wrong fins direction. So won't work with air flow from bottom to top. These are typical 3U enclosures where board fits inside in the inner slot and the enclosure fits between 3U rails. It needs different panel. The hole distance is 85mm.
I can't find this dimension in the datasheet. But once you mount it to the back of PCB, won't be able to fit into the rack.

**Comment by gkasprow** [comment]
@hartytp 15x15mm TEC, 2.5Ohm, 7W + 12.5mm heatsink fit perfectly
![obraz](https://user-images.githubusercontent.com/4325054/35412909-13fa1212-021e-11e8-9dd2-2f7110f7110b.png)

**Comment by gkasprow** [comment]
TEC model [CP50441](https://www.mouser.com/ds/2/670/cp50-1153086.pdf)

**Comment by jordens** [comment]
Just fine for forced horizontal airflow.
And AFAICT they will fit into the rack perfectly when the pcb is mounted to them. You only need a different front panel and card guide position.

**Comment by hartytp** [comment]
@gkasprow TEC and heat sink look good!

**Comment by gkasprow** [comment]
layout done, fixing some minor power integrity issues

**Comment by gkasprow** [comment]
Themal simulation with parameters:
![obraz](https://user-images.githubusercontent.com/4325054/35477218-fe8df50e-03be-11e8-8ac7-262e88c534f6.png)


![obraz](https://user-images.githubusercontent.com/4325054/35477216-f240ae22-03be-11e8-8d68-333b466cec26.png)

**Comment by gkasprow** [comment]
@hartytp DAC temp is 53 degrees without airflow
When I specify it as -0.6W instead of +400mW, the temperature drops to 42.1 degrees.
With airflow 20cm/s:
temp is 31.7 with TEC cooling and airflow and 38.5 without TEC cooling and with airflow.

**Comment by gkasprow** [comment]
Temperature distribution with 20cm/s airflow
![obraz](https://user-images.githubusercontent.com/4325054/35477278-dc324aae-03bf-11e8-8a47-ef1af204335b.png)

**Comment by gkasprow** [comment]
Current density for 12V rail. Far below limit.
![obraz](https://user-images.githubusercontent.com/4325054/35477341-e0d60e32-03c0-11e8-936d-db2226f9f988.png)

And for -12V rail. After optimisation:
![obraz](https://user-images.githubusercontent.com/4325054/35477382-921e3a0c-03c1-11e8-9721-d23d9054898b.png)

**Comment by a-shafir** [comment]
Can be improved:
1. C12.2 is connected so SIGGND but not AGND or ground plane.
2. J2.3 trace (and especially the via) looks weak
3. TEC1- pad clearance to SDO trace looks dangerous, can be short with hand soldering of TEC and burn the DAC.
4. Will TEC be powered from P3V3_MP? One pin of the signal cable IMHO it is not a good source for TEC power. Also it is shared power. Also it can be a long cable in some installations etc. Note: local P3V3 is also only 500mA max.

**Comment by a-shafir** [comment]
@gkasprow
Do you have any estimation of the TEC current needed?
With this layout the thermal impedance of the ground plane shall be low. 
I know you was trying to isolate the DAC thermally in V1.0. But not it is not isolated at all. So it looks like the TEC will work cooling via quite low thermal impedance to the hot board.
Can you add the "source of cold" to your simulation?

**Comment by gkasprow** [comment]
@a-shafir 
Can be improved:

> 
>     C12.2 is connected so SIGGND but not AGND or ground plane.
good catch, fixed

>     J2.3 trace (and especially the via) looks weak
OK, fixed, all current was passing via single pad

>     TEC1- pad clearance to SDO trace looks dangerous, can be short with hand soldering of TEC and burn the DAC.
there is solder mask, but I fixed

>     Will TEC be powered from P3V3_MP? One pin of the signal cable IMHO it is not a good source for TEC power. Also it is shared power. Also it can be a long cable in some installations etc. Note: local P3V3 is also only 500mA max.

nope, P3V3_MP is only to supply the I2C-> UART converter
>

**Comment by gkasprow** [comment]
@a-shafir in case of TEC I did experiments on prototype board. It needs very little power to cool by a few degrees. It has big heatsink above which has 10K/W at 200LFM (1m/s). We need max 1W of cooling power. Look at the plot below:
![obraz](https://user-images.githubusercontent.com/4325054/35481227-cea9dd88-041f-11e8-8a30-13090b1c6e81.png)
Let's assume that the hot side is 50 degrees. We need max 10degrees of difference and max 1W of cooling power. This means that we need much less than 0.4A (approx 0.2A) at input voltage ~2V.
That's why I did thermal simulation to see how much 1W of cooling power will change the DAC temperature.
You are right - I simply placed negative power to the DAC instead of other side of the PCB.
So I modified the design and attached thermal PAD to the TEC mode.  Hyperlynx ignores components without pads. So at the moment I can model DAC power and TEC power.
So with air flow of 1m/s at which TEC heatshink is specified, with -1W of Q, I get 20.5deg at the TEC pad and 23 deg at the DAC case.
With Q=0 I get 26.5 at the TEC pad and 26.8 at the DAC.
When I lower the air flow to 0.2m/s, the DAC temp is 45.7 and TEC pad temp is 45.1
With Q=1W the DAC temp is 39.4 deg.
With Q=2W and 0.2m/s air flow we get: DAC temp = 33.3 deg
This is the worst case scenario, with all components dissipating their max power.
Remember one thing - the Peltier module to cool down by 10 degrees at Q=1W needs to have roughly 20deg of temp difference due to heatsik thermal resistance and 10 deg temp increase at 1W. So in our case with cold side at 30 deg and Q=1W we get heatsink temperature of 50 degrees.

**Comment by gkasprow** [comment]
The temperature sensor is placed as close as possible to the DAC case. It is also thermally connected with thermal pad using 6 vias. I cannot do much better.
Where do you want to put another temperature sensor?

**Comment by gkasprow** [comment]
We would need to add to the heatsing to protect it against operation without the air flow. In such case the regulator will try to cool down the DAC but without proper ventilation, the hot side will get too hot and will destroy the TEC and possibly DAC because TEC will start heating instead of cooling.
We can simply add thermal switch attached to the sink which will simply disconnect the TEC supply.

**Comment by hartytp** [comment]
@gkasprow Thanks for doing all that! Look really good.

Thanks @a-shafir for the comments.

> C12.2 is connected so SIGGND but not AGND or ground plane.

Yes, let's try to make sure that all of the various grounds around the DAC are tied as well as possible to a single solid ground plane.

> With Q=0 I get 26.5 at the TEC pad and 26.8 at the DAC.
When I lower the air flow to 0.2m/s, the DAC temp is 45.7 and TEC pad temp is 45.1
With Q=1W the DAC temp is 39.4 deg.
With Q=2W and 0.2m/s air flow we get: DAC temp = 33.3 deg


That all sounds good @gkasprow.

Remember that the anticipated use case for Zotino is in a lab with decent temperature control, so the ambient temperature should only fluctuate by a degree or two (mainly due to variable power consumption of other EEMs in the same rack). So, we only want our TEC to change the DAC temperature by a couple of degrees max. The max TEC current should be up to +-1.5A (limited by the MaximIC peltier driver) so this gives us plenty of head room.

The fact that we have so much headroom suggests that (from a thermal perspective) we can afford to increase the amount of ground plane surrounding the DAC.

**Comment by hartytp** [comment]
> Basically we need to put one more sensor to the bottom-right side of the board and control it for a specific thermal difference between the sensors.

I don't understand this point. We're aiming to control the DAC's absolute temperature, not its relative temperature.

One sensor should be fine.

**Comment by hartytp** [comment]
@gkasprow My feedback (other than the final issues above that remain outstanding):
- [ ] Should we have a pin or two on the IDC to supply 12V power? My thinking here is that it would be nice for Thermostat be be able to operate in a mode where it's basically just a simple extension of Zotino, with power and I2C supplied via the cable.
- [ ] Why is the heatsink "H6?" (why the question mark?)
- [ ] If there is space to do it easily, shall we put test points on busy, LDAC, etc?
- [ ] I haven't looked at the layout in detail, but obviously please do a final sanity check before manufacture. e.g. is all the grounding (particularly around the DAC/reference) as solid as possible. Has the routing been done to minimise cross-talk, etc.
- [ ] What is C62 in terms of dielectric, voltage rating, etc? We should make sure this is a COG/NP0 with a decent voltage rating (in a decent size) to avoid excessive distortion. For components like this where the dielectric matters, please annotate that on the schematic (also applies to C61 and C56)
- [ ] Should C129 be tantalum/have a tantalum in parallel with it (ensures that capacitance doesn't drop off due to the voltage and provides some damping to stop the LC ringing).
- [ ] Add a note in the power budget that this does not include +-12V current consumed by the TEC driver, and that max current draw from that pin should be ?mA. Also add a "?mA max from +12V" annotation by the IDC connector on the schematic (replace ? with some sensible number).
- [ ] In the power budget please can you give the +12V current draw as well as the power (this makes it easier for users to add up the current draw from the +12V rail for all of their EEMs to check it's below the max that can be supplied by Kasli).

**Comment by hartytp** [comment]
Other than that, I'm happy for this to go to manufacture as it is.

**Comment by hartytp** [comment]
> We would need to add to the heatsing to protect it against operation without the air flow. In such case the regulator will try to cool down the DAC but without proper ventilation, the hot side will get too hot and will destroy the TEC and possibly DAC because TEC will start heating instead of cooling.
We can simply add thermal switch attached to the sink which will simply disconnect the TEC supply.

@gkasprow We should be able to solve this in software on the temperature controller (Thermostat). e.g. one can set a max current and a shutoff temperature in the controller software. That should be sufficient to prevent damage without having to add extra thermal switches etc. Let's not overcomplicate the hardware for this.

**Comment by a-shafir** [comment]
@hartytp imagine we have the DAC not powered and rest of the board powered. So we will have let's say 28C the board temperature under the DAC.  With the DAC powered it will be higher. So can compensate by the TEC the exceed heat from the DAC. In case if we will try to set the DAC 25C than there will be a temperature gradient so part of the board heat will be "pumped" with TEC. It looks like the ground plane thermal impedance us low. Greg can simulate how many W need to drop the temperature of the DAC few degree below the board temperature.
And it is not impossible to have the board temperature 30-40C or even more in some hot places or with insufficient airflow.
Any way, trying to drop the DAC temperature few deg under the surrounding PCB temperature will require high wattage of TEC and will make the situation even worst when the airflow is not good.
So if we have 1 or 2 (diagonal across) the DAC sensors we can cool the DAC right to the board temperature with the low TEC wattage as @gkasprov planned.

In case if we need to cool it below the board temperature need to check the thermal impedance and the extra TEC power needed etc.

**Comment by gkasprow** [comment]
@a-shafir we can always set the temperature of the DAC a few degrees below the ambient.
We don't care about absolute value but stability in long term.
The conditions in the lab don't change that often.

**Comment by a-shafir** [comment]
@gkasprov yes, if we will follow the board temperature and if the airflow will be that high than will need around 1W. Than it makes sense to power the thermostat from 12V of course. But for this need more sensor(s) - see above message and also some more advanced thermostat logic or hw.

**Comment by gkasprow** [comment]
@a-shafir the regulator we will use supports only one sensor.
We can add LM75 sensor on the board to sense ambient temperature. Then the regulator can be adjusted a few degrees below ambient.

**Comment by a-shafir** [comment]
@gkasprov what is the model of the regulator planned?

**Comment by gkasprow** [comment]
https://www.thorlabs.de/newgrouppage9.cfm?objectgroup_id=10626

**Comment by hartytp** [comment]
@a-shafir I'm not too worried about all of that.

Let's just stick with a single sensor rather than trying to overcomplicate all of this. Remember, the DAC doesn't have to be stabilised to mK, just to hundreds of mK in a rack that should be stable to a few K. Thus, we are only looking for a factor of 10 or so improvement in the ambient stability so we can ignore all second order effects.

**Comment by a-shafir** [comment]
@gkasprov my concern is not only temperature in the lab. If the airflow will not be sufficient the temperature inside of the case will slowly increase. Normally it finally (in hours) stabilizes because the efficiency of the cooling is increases with the device temperature (comparing to the ambient air). But in case of TEC if we are trying to hold a target temperature there is additional factor of the TEC hot side exceed power. So it is not impossible even a thermal runaway.

Also there are labs in hot places. Or what will happens if an unattended lab the air conditioning fails? If there is no target to keep the DAC at specific temperature indeed than there is certainly  way to stay with low power on TEC using note sensors and also limit TEC power etc.

**Comment by gkasprow** [comment]
@a-shafir so we can add brutal switch mounted on the heatsink that will brake the circuit.

**Comment by hartytp** [comment]
Currently, several of our designs will break without airflow (cf Kasli, unless the heatsink will fix that).

I'm not against a thermal switch if there is a really cheap, compact SMT option. But, I'm concerned that the complexity of this is the thing that's in most danger of going into runaway. We can always limit TEC V/I etc in software.

**Comment by a-shafir** [comment]
The protection is good to have.

Adding one more sensor cost nothing. In case of a programmable regulator it will allow to set the target temperature according to the situation.

You have all the data to decide.

**Comment by a-shafir** [comment]
"We can always limit TEC V/I etc in software." 

Basically it even does not sounds crazy just to power TEC from a fixed wattage source! It will just do the job.

**Comment by hartytp** [comment]
> Adding one more sensor cost nothing.

Well, you still have to pay someone to manually glue it to the heat sink and then wire it up to the PCB. Plus you have to sort out a comparator + thermal switch for the peltier. It all just seems like more hassle than it's worth IMHO.

Looking at Greg's simulations, the TEC power should be limited to about 1W by the TEC driver (which has user settable current and voltage limits for the Peltier). With a decent heatsink, that's not going to damage the TEC even if there isn't proper airflow (it's heatsink to the PCB on one side and a heatsink on the other). So, I just don't see this ever leading to the TEC breaking so long as the temperature controller is configured properly.

**Comment by a-shafir** [comment]
Since we have a lot of options about the gateware having the heatsink on FPGA the must IMHO. As well as some thermal protection HW for the part that does not have it internally.

**Comment by hartytp** [comment]
I can't parse that sentence. 

Look, let's keep this simple for the current version. If we have exploding TECs then we can add some extra protection.

In any case, we will probably want the TECs for this to get very good stability, but is this a niche use-case or will others (@jordens @dhslichter etc) use it as well?

**Comment by a-shafir** [comment]
"Well, you still have to pay someone to manually glue it to the heat sink and then wire it up to the PCB" for tracking the PCB temperature need a smd sensor on the PCB as I wrote initially. For TEC protection assuming we are not trying to lower the DAC temperature way below the PCB level.limiting the TEC power by 2w (for instance) shall be safe.

**Comment by hartytp** [comment]
What if we agree to just add a note on the schematic saying "2W max" by the IDC? Will that make everyone happy?

**Comment by a-shafir** [comment]
Note: "2W max" (and even 5W probably) will allow you to stabilize the DAC temperature assuming the rack temperature settled. But most likely it will be not enough to cool the DAC way below the board temperature because the major sources of hear around the DAC settled on the shared ground plane. So you still need yo know the board temperature to set the target for the regulator. So you need one more sensor on PCB. I2C sensor will work.

**Comment by hartytp** [comment]
Yes, one needs to know the ambient temperature, but one always needs to know that (it's generally not a good idea to pick a random temperature and rely of a lot of TEC power to reach it). However, that does not need a second sensor. The idea would be for the user to power the board with the TEC turned off, use the temperature controller (Thermistor) to measure the temperature and then use that as the set point.

Let's just leave the hardware as is and not bother with extra sensors.

**Comment by a-shafir** [comment]
Yep, with the i2c programmable TEC controller that allows to read the sensor it will work.

**Comment by jordens** [comment]
@gkasprow 
* Does that heatsink-TEC stack fit into a 4HP slot?
* Is there no point in putting an LDO in front of the P3V3 to the DAC digital section? Isolation good enough in the DAC?
* Nor a choke on DVCC of the DAC to decouple it a bit from the other chips on that rail?
Looks good otherwise!

@hartytp What stability and temperature coefficient do you expect with and without the TEC?

**Comment by gkasprow** [comment]
@jordens Yes, it fits in 4HP
I already added LC filter at the DAC P3V3

**Comment by hartytp** [comment]
> What stability and temperature coefficient do you expect with and without the TEC?

Target is <10ppm rms fluctuations (noise/drift) over 24 hours when used in a euro rack with few K fluctuations in temperature.

My memory of doing a detailed analysis of the stability of this board was that the stability is limited by the DAC (5ppm/K FSR typical)+ reference temp cos. Feeling was that it won't be quite good enough in a rack without some basic temperature control, but nothing particularly ambitious is needed.

Without the TEC, I'd expect something like 20ppm typical fluctuations with temperature (YMMV with conditions you mount the board in). With stabilisation, I expect few ppm stability easily (we've characterised a similar circuit based on these DACs before, and the stability was excellent once we added basic temperature stabilisation).

**Comment by jordens** [comment]
@hartytp In other words, temperature control gets you from 1 LSB stability to 0.1 LSB stability. I know where that's useful but the fact that quantization errors are now dominant by an order of magnitude definitely makes this a non-standard feature to me.

**Comment by hartytp** [comment]
@jordens yes. But, then so is the very low noise level that this board is targeting. It's also not that non-standard in the context of applying voltages to ion traps, where one often cares much more about stability than the exact value of the voltage.

**Comment by hartytp** [comment]
@gkasprow did you manage to finish off the items at the top of this issue (can we close it)? In particular, did you zero-index the signals? The schematic can be done later if that helps, but it would be good to apply it to the silk + FP before manufacture?

**Comment by gkasprow** [comment]
In Altium you cannot count multi-channel designs from 0. So one bus is counted from 1, but all other things are zero-indexed. PCB marking is also 0-indexed.
One can make Altum desing 0-indexed but then we would have to have 32 copies of the filter block on the schematic page

**Comment by gkasprow** [comment]
Yes, you can close it.

**Comment by hartytp** [comment]
@gkasprow uuurgh, nasty!

Okay, the current situation is fine. However, it's a little confusing that the schematics don't match the silkscreen of front panel. Maybe add a one line annotation at the top of the schematic near the connectors saying something like "Note: DAC channels are zero-indexed on silk screen and front panel"?

Can you also upload the files to a release, please?

Also, did you fix the screening enclosure to fit the TEC + heatsink?

**Comment by hartytp** [comment]
Thanks again for the great work on this Greg!

**Comment by gkasprow** [comment]
once PCB house accepts them, I will upload.

**Comment by hartytp** [comment]
@gkasprow One thing I think we got wrong in this version is the ground return for the 12V TEC current. This should really have gone back to the EEM connector with a dedicated thick ground trace rather than returning through the main ground (it might also have been worth adding a capacitor by the TEC IDC).

Normally I don't like separating grounds like this, but DC power traces are the one place where I it does work well, and the TECs can draw enough current to cause measurable changes in the DAC's output voltage.

In practice, this may not end up being a problem, since it looks like we will not use the +12V or I2C from this header, as Thermostat will have ethernet + PoE....

**Comment by gkasprow** [comment]
@hartytp indeed there should be return path that does not pass via our DAC circuit. But the current consumption at 12V won't be high - max 300mA and the return current path is quite far away from the ADC. There are already cut-outs in the ground plane just to limit the return path of the SMPS so I don't afraid of them. I can make simulation to check how this return current propagates.

**Comment by hartytp** [comment]
@gkasprow Let's not worry about it for now. It's easy to fix in the next revision.

**Comment by a-shafir** [comment]
I added BOM concerning issue in the errata thread.

**Comment by gkasprow** [comment]
Don't worry about 0.1% resistors. We have plenty of them on stock and their price impact is negligible.

**Comment by a-shafir** [comment]
@gkasprow as far as I understand from the previous discussion there are a lot of efforts to do right things, not to make it too complicate or "without a reason". For usd0.23 * 64  can be added even another DAC on the board...

Any way IMHO there shall be a way for mass production. If for the production optimization will need a lot of BOM changes it will be a risk of unexpected issues, so better to keep it tight.

I see 3 options:
1. Put it in errata or to do list " replace xx to 1%"
2. Put this part on the BOM and sch as generic 1% but in the assembly BOM you can choose 0.1% as "stock".
3. Keep that reels of 0.1% in stock and use it in the next designs where matched resistors are the must (instrumental amplifiers, bridges etc). The values are vell suitable for this.

Option  3 is the most rational IMHO and just ignoring it is the least rational. Just my opinion.

**Comment by gkasprow** [comment]
We already have rails of mostly 0.5 and 0.1% resistors. It does not make sense to keep stock of 5, 1, 0.5 and 0.1% in 0402, 0603, 0805 sizes because it would cost fortune. It's cheaper to keep stock of only 0.1% resistors. And we buy 100,000 pcs, not 100 pcs.

**Comment by gkasprow** [comment]
Once somebody wants to produce these boards, he can change a few lines of BOM with suitabe values.

**Comment by a-shafir** [comment]
@gkasprow what is the price for 0.1% at your source? I also would like to have such source that I will be able to use 0.1% freely. For most of my previous designs targeted mass production's keeping the specs AND the target cost was the must.

Regarding the BOM. As I wrote. Normally in the boms there is a column with the basic specs like R0402_1k_1% it can be more detailed when it is not so generic like R0402_1k_1%_150mW or it can be a specific part number - that minds ONLY this part meets the board needs (like Tyco RN73C1J2K32BTDF). This is the "design BOM. 

There is also a production BOM that have a specific vendor parts or even a factory stock locator.

Since we have already hundreds of positions in the BOMs (looking all the boards active) it will be more and more difficult to order non generic " irreplaceable " parts. So better to start some CIS system.

Otherwise for the further production the design engineer must review and approve EVERY replacement of the non-generic part! So better to have non-generic only where it need indeed like the key components.

Of course this system can be changed later. With way more efforts. 

This is besides the 3rd party production matters you mention.

IMHO

**Comment by hartytp** [comment]
>  the must

"*a* must"?

**Comment by hartytp** [comment]
@a-shafir While I appreciate your enthusiasm about this, @gkasprow is the lead engineer here and this kind of decision is up to him. You're free to raise issues like this, but once you've been overruled please drop it.

I don't want to see this being a long argument about ways of shaving 1% off the cost of manufacturing these boards -- and even if we did want to do that I suspect that there are lower hanging fruit than worrying about resistor tolerances.

**Comment by gkasprow** [comment]
@a-shafir For all my designs i use same, unified library just to make sure that all the components the assembly house has on stock.
They know I use same library and know that they can invest and buy whole reels of these components.
I have and use 1% resistors for certain popular values. But rarely used like E192 values I have in my lib only 0.1% just to make life easier and keep the total cost low. Otherwise I'd have to keep several tolerances of  every E192 resistor.
Take into account that every new BOM line is costly to maintain in the assembly house, it also means that would need to be ordered in low quantities which is simply not worth the effort.
We produce thousands of other boards for many different clients and this is the best way we managed to figure out during last ten years.

**Comment by a-shafir** [comment]
@hartytp OK, no worries. In this case I am concerning only of streamlining the workflow. Also since the project is "public" keeping it more "friendly" to produce and just clean will help the sayma ecosystem. 

There is nothing personal. I don't want to make one "wrong" or myself "right" etc.

About the labeling 500r sounds for me more important since a dypically non-rf DAC voltage output is buffered and some current limit labeled. I clearly understand why adding 32 buffers is a bad idea here but adding the label line shall help.

**Comment by a-shafir** [comment]
@gkasprow OK, than it is just labeling issue on the scematics. 
Indeed there is no easy way to solve this readability of sch and linkage to the BOM.
Showing all attributes of the parts makes sch non-readable. Showing only value is leading for very scrupulous BOM check.
It looks for me the most cleat way that allows printable schematics but still very clear is to list some "defaults" on one of the page like "resistor: 0402, 5%, generic; capacitor: 0402 COG 5% 50V..."
Than show all non-default attributes like "10k", " 10k_0603", "10uf_0603_x7r" or even "RN73C1J2K32BTDF".
How it sounds? 
Note:I just think that establishing this or any better system for the new designs if it will help.

**Comment by gkasprow** [comment]
You can use Altium viewer to access full details. Long time ago I used to enable footprint and PN but schematic became a mess.
That's why you have a BOM and CTRL+F command to investigate. And free Altium Viewer.

**Comment by a-shafir** [comment]
I wish it to be illustrative and visible
Something like this
https://github.com/m-labs/spectropak/blob/master/hvamp_proto3/spectroamp3.pdf
I can't say there is one pattern that works for all
In some cases even a netlist can be more readable.

**Comment by gkasprow** [comment]
I have such parameters that can be enabled
![obraz](https://user-images.githubusercontent.com/4325054/35541563-bb82e17c-055b-11e8-9a0e-6a88873b83ad.png)

**Comment by a-shafir** [comment]
I also tried various layouts.
There is drawback of any. Adding more data to sch at least needs bigger paper to print (keeping it readable).
The visible value is easiest to use since there is no toyaly strict rules for it. And it is one line.
Showing all attributes for every parts in one string does not work - your example is great.
My idea is:
1. have some "defaults" (per schematic)
Can be printed or " common sense". 
Like <1000pf is COG <1uf X7R etc.

2. add to the value field manually all non-common attributes.

3. Use part number when the part can't be replaced easily.

Basically there shall be is a way to do it automatically by a CAD plugin. .re critic

**Comment by gkasprow** [comment]
I can enable any combinations of properties I've shown you just with one click.
so OK.
For resistors only case makes sense because tolerance is already printed on the symbol.
For capacitors I can enable dielectric only where it is essential. Rest we simply assume X5R/X7R.
It does not make sense to enable any additional parameters for resistor supplying LED

**Comment by a-shafir** [comment]
@gkasprow OK. I think I see an elegant solution for this!
What shall be done is literally packing the non-readable very long string (refer to the example you captured) by a simple pattern (literally a RegExp or similar).
It can be done by a script.
I will try to make such script that will work with the altum files.

How it sounds?

**Comment by a-shafir** [comment]
@gkasprow your idea is also working. Of course need to show it smart and to the point exactly when you need to use ctrl-f to verify the attribute is not missing.
Doing this automatically by a tool is more robust because it will pop-up any "unexpected" attributes like the copy-paste errors etc.

**Comment by gkasprow** [comment]
I'm not going to copy and paste anything. I can simply enable visibility of certain parameters in the places that they are essential.
